src/main/java/net/neoforged/neoforge/common/conditions/ElementExistsCondition.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.conditions;
+
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.Holder;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+
+public record ElementExistsCondition<TRegistry>(ResourceKey<TRegistry> registryKey) implements ICondition {
```

Comment:
I don't like the name "element" which is not used at all in other contexts referring to registry entries.

Reply:
Idea: `RegisteredCondition`? And replace `element` by `value`? 🤔